### PR TITLE
Add an `IndexedMap` article (without `MultiIndex`)

### PR DESCRIPTION
**Pull request by uint**
1st draft of the `IndexedMap` article.

TODO in follow-ups:
- [ ] link to an expanded blurb about compound key wonkiness (https://github.com/CosmWasm/docs/issues/51)
- [ ] write a section about `MultiIndex` (https://github.com/CosmWasm/docs/issues/79)

### CODE REVIEW

File: src/pages/cw-storage-plus/containers/indexed-map.mdx
Comment (dakom) [inline src/pages/cw-storage-plus/containers/indexed-map.mdx:62]: I don't think I've used this attribute before, but I haven't touched IndexedMap in a long time. Mind adding a couple notes about what it does?

File: src/pages/cw-storage-plus/containers/indexed-map.mdx
Comment (dakom) [inline src/pages/cw-storage-plus/containers/indexed-map.mdx:68]: hehe.. so... the example here looks a closure. In reality though, it's a function pointer, it can't really capture anything from the environment.

on the one hand, that's a very nuanced and confusing advanced Rust topic, definitely not worth calling out (that the closure typechecks as a regular `fn` because the compiler sees it's not capturing anything)

on the other hand, I was personally tripped up by this when I wanted to build the index out of the provided data and something else loaded from state (I'll neither confirm nor deny that I used a global mutable static var to get around that :P)

maybe a note with something like "we're using a closure here for convenience, but it's really a regular rust function pointer - as long as it doesn't capture anything it will work". Don't love that wording though. Your call, maybe worth kicking around a bit?

File: src/pages/cw-storage-plus/containers/indexed-map.mdx
Comment (dakom) [inline src/pages/cw-storage-plus/containers/indexed-map.mdx:64]: how do the type arguments here relate to the index? i.e. `Addr` was noted as being primary... how does that relate to the signature here, is primary always the last?

File: src/pages/cw-storage-plus/containers/indexed-map.mdx
Comment (dakom) [inline src/pages/cw-storage-plus/containers/indexed-map.mdx:187]: I was a _bit_ confused by this - saw "alice" and thought maybe we're looking up the address... totally my mistake, but I'd suggest for the sake of example to give alice and bob different strings for the address vs. handle

File: src/pages/cw-storage-plus/containers/indexed-map.mdx
Comment (dakom) [inline src/pages/cw-storage-plus/containers/indexed-map.mdx:201]: whoa! I never knew you could do that (partial bounds)! Makes sense, but TIL :)

**Review by dakom — APPROVED** [review]
I left a few comments, but none of them should block the PR from my perspective :)

**Review by hashedone — APPROVED** [review]
Good one! Left some comments on vercel, most of them for you to decide.

File: src/pages/cw-storage-plus/containers/indexed-map.mdx
Comment (uint) [reply src/pages/cw-storage-plus/containers/indexed-map.mdx:62]: Sure, can do

File: src/pages/cw-storage-plus/containers/indexed-map.mdx
Comment (uint) [reply src/pages/cw-storage-plus/containers/indexed-map.mdx:68]: Oh, huh. Yeah, I'll change it to something that's not confusing

File: src/pages/cw-storage-plus/containers/indexed-map.mdx
Comment (uint) [reply src/pages/cw-storage-plus/containers/indexed-map.mdx:187]: Yeah, good call

File: src/pages/cw-storage-plus/containers/indexed-map.mdx
Comment (uint) [reply src/pages/cw-storage-plus/containers/indexed-map.mdx:64]: Yeah, primary is last. Do you think it should be noted in the docs?

Comment (dakom) [reply src/pages/cw-storage-plus/containers/indexed-map.mdx:64]: yeah, I think it's worth noting

**Pull request merged**
